include/pybind11/detail/common.h

```diff
@@ -436,14 +436,14 @@ PyModuleDef_Init should be treated like any other PyObject (so not shared across
         PYBIND11_CHECK_PYTHON_VERSION                                                             \
         pre_init;                                                                                 \
         PYBIND11_ENSURE_INTERNALS_READY                                                           \
-        static ::pybind11::detail::slots_array slots = ::pybind11::detail::init_slots(            \
+        static ::pybind11::detail::slots_array pybind11_slots = ::pybind11::detail::init_slots(   \
```

Comment:
`pybind11_slots` could be surprising later (send intuition on the wrong track).

Maybe: `mod_def_slots`?